include/LightGBM/utils/common.h

```diff
@@ -36,9 +36,17 @@
 #elif MM_MALLOC
 #include <mm_malloc.h>
 #elif defined(__GNUC__)
-#include <malloc.h>
-#define _mm_malloc(a, b) memalign(b, a)
-#define _mm_free(a) free(a)
+  // https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html
+  // https://www.oreilly.com/library/view/mac-os-x/0596003560/ch05s01s02.html
+  #ifdef HAVE_MALLOC_H
+    #include <malloc.h>
+    #define _mm_malloc(a, b) memalign(b, a)
+    #define _mm_free(a) free(a)
+  #else
+    #include <stdlib.h>
+    #define _mm_malloc(a, b) malloc(a)
+    #define _mm_free(a) free(a)
+  #endif
```

Comment:
It seems that this can be simplified to not duplicate the part with `<stdlib.h>`:
```
#elif defined(__GNUC__) && defined(HAVE_MALLOC_H)
  #include <malloc.h>
  #define _mm_malloc(a, b) memalign(b, a)
  #define _mm_free(a) free(a)
#else
  #include <stdlib.h>
  #define _mm_malloc(a, b) malloc(a)
  #define _mm_free(a) free(a)
#endif
```

Reply:
ok

Reply:
fixed in https://github.com/microsoft/LightGBM/pull/3188/commits/04906c5d846f42d2f50ec24ca28675b6e8bf4cdb